src/ray/core_worker/core_worker.cc

```diff
@@ -195,6 +195,7 @@ CoreWorker::CoreWorker(const WorkerType worker_type, const Language language,
         // behaviour. TODO(ekl) backoff exponentially.
         RAY_LOG(ERROR) << "Will resubmit task after a 5 second delay: "
                        << spec.DebugString();
+        absl::MutexLock lock(&mutex_);
```

Comment:
I believe this is a possible thread-safety issue since we pass this callback into TaskManager.

Reply:
Any chance of deadlocking with the task manager here if it can now make a callback that takes the core worker locK?

Reply:
It's a future risk, but right now we don't call back into task manager while holding the lock.